### PR TITLE
chapter 11 email activation

### PR DESCRIPTION
**Pull request by dungbka**

### CODE REVIEW

File: app/controllers/account_activations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class AccountActivationsController < ApplicationController
+  def edit
+    user = User.find_by(email: params[:email])
```
Comment (oPhamThiDao) [inline app/controllers/account_activations_controller.rb]: bo ()

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -18,9 +18,9 @@ def create
     @user = User.new user_params
 
     if user.save
-      log_in user
-      flash[:success] = t "welcome_to_app"
-      redirect_to user
+      user.send_activation_email
+      flash[:info] = "Please check your email to activate your account."
```
Comment (oPhamThiDao) [inline app/controllers/users_controller.rb]: I18n

File: app/mailers/user_mailer.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class UserMailer < ApplicationMailer
+  def account_activation user
+    @user = user
+    mail to: user.email, subject: "Account activation"
```
Comment (oPhamThiDao) [inline app/mailers/user_mailer.rb]: I18n

File: config/environments/development.rb [hunk 4]
```diff
@@ -13,12 +13,12 @@
...
 
   # Enable/disable caching. By default caching is disabled.
-  if Rails.root.join('tmp/caching-dev.txt').exist?
+  if Rails.root.join("tmp/caching-dev.txt").exist?
     config.action_controller.perform_caching = true
 
     config.cache_store = :memory_store
     config.public_file_server.headers = {
-      'Cache-Control' => "public, max-age=#{2.days.seconds.to_i}"
+      "Cache-Control" => "public, max-age=172800"
```
Comment (oPhamThiDao) [inline config/environments/development.rb]: "Cache-Control": "public, max-age=172800"

File: config/environments/test.rb [hunk 5]
```diff
@@ -15,8 +15,9 @@
   # Configure public file server for tests with Cache-Control for performance.
   config.public_file_server.enabled = true
   config.public_file_server.headers = {
-    'Cache-Control' => "public, max-age=#{1.hour.seconds.to_i}"
+    "Cache-Control" => "public, max-age=#{1.hour.seconds.to_i}"
```
Comment (oPhamThiDao) [inline config/environments/test.rb]: "Cache-Control": "public, max-age=#{1.hour.seconds.to_i}"

File: config/environments/test.rb [hunk 6]
```diff
@@ -15,8 +15,9 @@
...
   }
+  config.action_mailer.default_url_options = { host: "example.com" }
```
Comment (oPhamThiDao) [inline config/environments/test.rb]: {host: "example.com"}

File: config/environments/test.rb [hunk 7]
```diff
@@ -15,8 +15,9 @@
...
 
   # Show full error reports and disable caching.
   config.consider_all_requests_local       = true
```
Comment (oPhamThiDao) [inline config/environments/test.rb]: config.consider_all_requests_local = true

File: app/models/user.rb [hunk 8]
```diff
@@ -38,18 +39,33 @@ def remember
...
+    return false if digest.blank?
+    BCrypt::Password.new(digest).is_password?(token)
   end
 
   def forget
     update_attributes remember_digest: nil
   end
 
+  def activate
+    update_attributes activated: true
```
Comment (dieptx) [inline app/models/user.rb]: Viết chung 1 dòng được mà

File: app/views/user_mailer/account_activation.html.erb [hunk 9]
```diff
@@ -0,0 +1,8 @@
+<h1><%= t "mailer.sample_app" %></h1>
+
+<p><%= t "mailer.hi" %> <%= @user.name %>,</p>
+
+<p>
+<%= t "mailer.welcome" %>
```
Comment (dieptx) [inline app/views/user_mailer/account_activation.html.erb]: ident

File: app/views/user_mailer/account_activation.text.erb [hunk 10]
```diff
@@ -0,0 +1,4 @@
+<%= t "mailer.hi" %>
+<%= @user.name %>,
+<%= t "welcome" %>
+<%= edit_account_activation_url(@user.activation_token, email: @user.email) %>
```
Comment (dieptx) [inline app/views/user_mailer/account_activation.text.erb]: Bỏ ()

**Pull request merged**